File: SuperMovie_server/src/api/User/AddUser.cs
namespace SuperMovie_server.api.User;

using WebSocketSharp;
using WebSocketSharp.Server;

public struct AddUserReq
{
    public long UserId;
    public long UserPwd;
}

public struct AddUserRsp
{
    public bool Ok;
}

//api : add_user
public class AddUser : WebSocketBehavior
{
}

File: movieui/src/scripts/ws/User/addUser.ts
import {wsUrlRoot} from "@/scripts/ws/meta"
import {createWebSocket, recvMsg, reqStringify, rspParse, sendMsg} from "@/scripts/ws/helper"

export {addUser}
export type {
    AddUserReq,
    AddUserRsp
}

type AddUserReq =
    {
        UserId: bigint
        UserPwd: string
    }

type AddUserRsp =
    {
        Ok: boolean
    }

async function addUser(req: AddUserReq) {
    const conn = createWebSocket(`${wsUrlRoot}/add_user`)

    const task = recvMsg(conn)
    const reqJson = reqStringify(req)
    console.log('add_user req:' + reqJson)
    sendMsg(conn, reqJson).then()
    const msg = await task
    console.log('add_user rsp:' + msg)

    return <AddUserRsp>rspParse(msg)
}